src/v/config/configuration.cc

```diff
@@ -1630,6 +1630,18 @@ configuration::configuration()
       {.visibility = visibility::user},
       std::nullopt,
       &validate_non_empty_string_opt)
+  , cloud_storage_url_style(
+      *this,
+      "cloud_storage_url_style",
+      "The addressing style to use for S3 requests.",
+      {.needs_restart = needs_restart::yes,
+       .example = "virtual_host",
+       .visibility = visibility::user},
+      cloud_storage_clients::s3_url_style::virtual_host,
+      {
+        cloud_storage_clients::s3_url_style::virtual_host,
+        cloud_storage_clients::s3_url_style::path,
+      })
```

Comment:
Should we add an `auto` option? From what I can tell the strategy is basically to try virtual-host and then fallback to path-style, and cache the decision.

See https://github.com/boto/botocore/blob/develop/botocore/utils.py#L1375-L1381

@Lazin what do you think?

@WillemKauf if we all think this is a good idea, let's create a new jira ticket to capture it--no need to expand the scope of this PR.

Reply:
I think this is a great idea for future work! +1